swarm/src/lib.rs

```diff
@@ -475,23 +475,15 @@ where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
         &mut self.behaviour
     }
 
-    /// Returns the next event that happens in the `Swarm`.
-    ///
-    /// Includes events from the `NetworkBehaviour` but also events about the connections status.
-    pub async fn next_event(&mut self) -> SwarmEvent<TBehaviour::OutEvent, THandleErr> {
-        future::poll_fn(move |cx| ExpandedSwarm::poll_next_event(Pin::new(self), cx)).await
-    }
-
     /// Returns the next event produced by the [`NetworkBehaviour`].
-    pub async fn next(&mut self) -> TBehaviour::OutEvent {
-        future::poll_fn(move |cx| {
-            loop {
-                let event = futures::ready!(ExpandedSwarm::poll_next_event(Pin::new(self), cx));
-                if let SwarmEvent::Behaviour(event) = event {
-                    return Poll::Ready(event);
-                }
+    pub async fn behaviour_next(&mut self) -> TBehaviour::OutEvent {
```

Comment:
If we want to keep this, I think `next_behaviour` would be easier to discover through auto-completion provided that users know that `Swarm` implements `Stream` and try `.next` first.

Reply:
That's a good point. The reason why I didn't chose the name was because it may be misleading/ confusing, but then again it should be clear what the method does once you look at the return type and docs. 
Or what do you think about `next_behaviour_event`/ some abbreviation of it?
But in general I am fine with either name.